### PR TITLE
Added custom render trigger on order item component.

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- added custom trigger for OrderItem render method
  - Expanding/Collapsing the order item was slow
  - When expanding every single order item was re-rendered
  - added direct connection to redux from each order item
  - added should component update trigger 
  - render times are down from 125 ms to 15 ms per order item

### CODE REVIEW

**Comment by gmcculloug** [comment]
Based on the description 👍 
@martinpovolny Please review

File: src/smart-components/order/order-item.js [hunk 1]
```diff
@@ -27,91 +28,100 @@ import { getOrderIcon, getOrderPortfolioName } from '../../helpers/shared/orders
 import { createOrderedLabel, createUpdatedLabel, createDateString } from '../../helpers/shared/helpers';
 import createOrderRow from './create-order-row';
 
-const OrderItem = ({ item, isExpanded, handleDataItemToggle, portfolioItems }) => {
-  const { finishedSteps, steps } = createOrderRow(item);
-  const orderedAt = createOrderedLabel(new Date(item.ordered_at));
-  const updatedAt = createUpdatedLabel(item.orderItems);
+class OrderItem extends Component {
+  shouldComponentUpdate({ isExpanded }) {
```
Comment (lgalis) [inline src/smart-components/order/order-item.js]: this could further be improved by returning true only when the item is being expanded, not when closed.

Comment (martinpovolny) [reply src/smart-components/order/order-item.js]: I do not follow.

**Comment by martinpovolny** [comment]
Looks good to me. Please, resolve the conflicts.

File: src/smart-components/order/order-item.js (hunk 1 above)
Comment (Hyperkid123) [reply src/smart-components/order/order-item.js]: If i understand your suggestion correctly it would mean the component would never close.

**Pull request merged**